### PR TITLE
Add ignore_if_coin_type_mismatch to restore_backup()

### LINKED ISSUE #1914: Restore client options only optionally from a backup
## Description

Restore client options only optionally from a backup

Also update the bindings

## Motivation

If a backup is restored and other client options want to be used, there is no need to update them from the backup

## Requirements

1. Add an `ignore_client_options_if_network_mismatch: Option<bool>` in `restore_backup()` which if set to Some(true) will compare the network name in the client options of the account manager and the backup and only updates them, if they have a different network name
2. Update message interface
3. Update bindings

## Open questions (optional)

Use this section to ask any questions that are related to the feature.

## Are you planning to do it yourself in a pull request?

Yes/No.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add ignore_if_coin_type_mismatch to restore_backup()

## Links to any relevant issues

Fixes https://github.com/iotaledger/wallet.rs/issues/1914

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Added test

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: wallet/src/message_interface/message.rs [hunk 1]
```diff
@@ -104,14 +104,18 @@ pub enum Message {
     /// Replaces client_options, coin_type, secret_manager and accounts. Returns an error if accounts were already
     /// created If Stronghold is used as secret_manager, the existing Stronghold file will be overwritten. If a
     /// mnemonic was stored, it will be gone.
-    /// Expected response: [`Ok`](crate::message_interface::Response::Ok)
+    /// if ignore_if_coin_type_mismatch.is_some(). client options will not be restored
```
Comment (thibault-martinez) [inline wallet/src/message_interface/message.rs]:
```suggestion
    /// if ignore_if_coin_type_mismatch.is_some(), client options will not be restored
```

File: wallet/src/account_manager/operations/stronghold_backup/mod.rs [hunk 2]
```diff
@@ -55,7 +55,15 @@ impl AccountManager {
     /// Replaces client_options, coin_type, secret_manager and accounts. Returns an error if accounts were already
     /// created If Stronghold is used as secret_manager, the existing Stronghold file will be overwritten. If a
     /// mnemonic was stored, it will be gone.
-    pub async fn restore_backup(&self, backup_path: PathBuf, mut stronghold_password: String) -> crate::Result<()> {
+    /// if ignore_if_coin_type_mismatch.is_some(). client options will not be restored
```
Comment (thibault-martinez) [inline wallet/src/account_manager/operations/stronghold_backup/mod.rs]:
```suggestion
    /// if ignore_if_coin_type_mismatch.is_some(), client options will not be restored
```

File: wallet/tests/backup_restore.rs [hunk 3]
```diff
@@ -192,3 +195,179 @@ async fn backup_and_restore_mnemonic_secret_manager() -> Result<()> {
...
+        .with_coin_type(SHIMMER_COIN_TYPE)
+        .finish()
+        .await?;
+
+    // restore with ignore_if_coin_type_mismatch: Some(true) to not overwrite the coin type
+    restore_manager
+        .restore_backup(
+            PathBuf::from("test-storage/backup_and_restore_same_coin_type/backup.stronghold"),
+            stronghold_password.to_string(),
+            Some(true),
```
Comment (thibault-martinez) [inline wallet/tests/backup_restore.rs:351]: Can you add a `Some(false)/None` example?

Comment (Thoralf-M) [reply wallet/tests/backup_restore.rs:351]: None exists already further above

**Review by thibault-martinez — APPROVED** [review]

**Review by Tuditi — APPROVED** [review]

**Pull request merged**